perses/rjmc/geometry.py

```diff
@@ -117,16 +117,22 @@ def propose(self, top_proposal):
 
                 #propose a bond and calculate its probability
                 bond = self._get_relevant_bond(atom, bond_atom)
-                r_proposed = self._propose_bond(bond, top_proposal.beta)
-                logp_r = self._bond_logp(r_proposed, bond, top_proposal.beta)
+                r_proposed = self._propose_bond(bond, beta)
+                bond_k = bond.type.k*units.kilocalorie_per_mole/units.angstrom**2
+                sigma_r = units.sqrt(1/beta*bond_k)
+                logZ_r = np.log((np.sqrt(2*np.pi)*sigma_r/sigma_r.unit))
+                logp_r = self._bond_logq(r_proposed, bond, beta) - logZ_r
```

Comment:
Not something to worry about optimizing here, but I think lines 120-124 end up retrieving `bond_k` and recomputing `sigma_r` a total of three independent times.  There may be some refactoring we could think of in future that cut down on code duplication.